src/utils/push.ts

```diff
@@ -0,0 +1,24 @@
+export interface Criteria {
```

Comment:
I would move it to a `/types` folder somewhere. And possibly with a different name:

```
// /types/member.types.ts

export interface Member {
  key: string;
  all: boolean;
  names: string[];
}

export type MembersMap = { [key: string]: Member };
```

MembersMap also must have type, it's being used in many places.